tests/unit/lifecycle/LifecycleTask.spec.js

```diff
@@ -261,4 +283,198 @@ describe('lifecycle task helper methods', () => {
+        'LastModified times', () => {
+            // Both arrays should be sorted respective to their own arrays.
+            // This method should stable sort and merge both arrays
+            const versions = [
+                {
+                    Key: 'obj-1',
+                    VersionId:
+                    '834373731313631393339313839393939393952473030312020353833',
+                    IsLatest: true,
+                    LastModified: '2018-04-04T23:16:46.000Z',
+                },
+                // LastModified matches with a delete marker
+                {
+                    Key: 'obj-1',
+                    VersionId:
+                    '834373731313631393339313839393939393952473030312020353830',
+                    IsLatest: false,
+                    LastModified: '2018-04-04T23:16:44.000Z',
+                },
+                {
+                    Key: 'obj-1',
+                    VersionId:
+                    '834373731313631393339313839393939393952473030312020353827',
+                    IsLatest: false,
+                    LastModified: '2018-04-04T23:16:41.000Z',
+                },
+                {
+                    Key: 'obj-1',
+                    VersionId:
+                    '834373731313631393339313839393939393952473030312020353823',
+                    IsLatest: false,
+                    LastModified: '2018-04-04T23:16:32.000Z',
+                },
+            ];
+            const dms = [
+                {
+                    Key: 'obj-1',
+                    IsLatest: false,
+                    VersionId:
+                    '834373731313631393339313839393939393952473030312020353831',
+                    LastModified: '2018-04-04T23:16:44.000Z',
+                },
+                {
+                    Key: 'obj-1',
+                    IsLatest: false,
+                    VersionId:
+                    '834373731313631393339313839393939393952473030312020353825',
+                    LastModified: '2018-04-04T23:16:34.000Z',
+                },
+            ];
+
+            // Can only do this since I set VersionId in the expected order
+            // Normally, when dealing with multiple objects, we wouldn't be able
+            // to sort by just VersionId, since they could be intertwined.
+            const expected = [...versions, ...dms].sort((a, b) => (
+                a.VersionId < b.VersionId
+            ));
+            const res = lct._mergeSortVersionsLists(versions, dms);
+
+            assert.deepStrictEqual(expected, res);
+        });
+    });
+
+    describe('_applyVersionStaleDate', () => {
+        const defaultVersions = {
+            ETag: '"6f5902ac237024bdd0c176cb93063dc4"',
+            Size: 0,
+            StorageClass: 'STANDARD',
+        };
+
+        it('should apply a staleDate property on each version on a list ' +
+        'of versions', () => {
+            const list = [
```

Comment:
Suggesting to add the second object key in the same list (`obj-2`), so that you can check that `staleDate` applies independently to each object.

Reply:
good suggestion, actually caught a bug :)